framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.java

```diff
@@ -105,6 +114,23 @@
   /** Whether the {@code -AinferOutputOriginal} option was supplied to the checker. */
   private final boolean inferOutputOriginal;
 
+  /**
+   * Returns the names of all qualifiers that are marked with {@link InvisibleQualifier}, and that
+   * are supported by the given type factory.
+   *
+   * @param atypeFactory a type factory
+   * @return the names of every invisible qualifier supported by {@code atypeFactory}
+   */
+  public static Set<String> getInvisibleQualifierNames(AnnotatedTypeFactory atypeFactory) {
+    return atypeFactory.getSupportedTypeQualifiers().stream()
+        .filter(
+            qual ->
+                Arrays.stream(qual.getAnnotations())
+                    .anyMatch(anno -> anno.annotationType() == InvisibleQualifier.class))
```

Comment:
My suggestion was to abstract out `isInvisible`, which might be useful in another context.
But, this is OK if you prefer it.